modules/programs/newsboat.nix

```
@@ -126,11 +133,11 @@ in {
     # Use ~/.newsboat on stateVersion < 21.05 and use ~/.config/newsboat for
     # stateVersion >= 21.05.
     home.file = mkIf (versionOlder config.home.stateVersion "21.05") {
-      ".newsboat/urls".text = urlsFileContents;
+      ".newsboat/urls" = mkIf cfg.declarativeUrls { text = urlsFileContents; };
```

Perhaps simply do
```suggestion
      ".newsboat/urls" = mkIf (cfg.urls != []) { text = urlsFileContents; };
```
instead? Then you don't need to add a special option for this, just update the `urls` option description.

---

This looks great, let's go with that.